src/poddlthread.cpp

```diff
@@ -120,8 +120,11 @@ void PodDlThread::run()
 			LOG(Level::DEBUG,
 				"PodDlThread::run: download complete, deleting "
 				"temporary suffix");
-			rename(filename.c_str(), dl->filename().c_str());
-			dl->set_status(DlStatus::READY);
+			if(rename(filename.c_str(), dl->filename().c_str()) == 0){
+				dl->set_status(DlStatus::READY);
+			}else{
+				dl->set_status(DlStatus::RENAME_FAILED,"Unable to rename downloaded file. Please check if you have sufficient permissions.");
```

Comment:
`rename()` can fail for a number of reasons, see its man page. Instead of a blanket message about permissions, it's better to invoke `strerror` which will return a more detailed message.

Reply:
Done